core/src/main/scala/akka/persistence/r2dbc/R2dbcSettings.scala

```diff
@@ -82,6 +82,11 @@ final class QuerySettings(config: Config) {
  */
 @InternalStableApi
 final class ConnectionFactorySettings(config: Config) {
+
+  val urlOption: Option[String] =
+    if (config.hasPath("url")) Option(config.getString("url"))
+    else None
```

Comment:
We don't use `null` config values in Akka, but instead `""` for an optional property like this. It would still be an Option in code. I don't have a strong opinion about that. Maybe ConfigChecker (finding typos) doesn't work with null values, I don't know.